src/seer_app.erl

```diff
@@ -4,6 +4,17 @@
 
 -export([start/2, stop/1]).
 
-start(_StartType, _StartArgs) -> seer_sup:start_link().
+start(_StartType, _StartArgs) ->
+    seer_sup:start_link(),
+    Poller =
+        #{
+            id => seer_server,
+            start => {seer_server, start_link, []},
+            shutdown => 2000,
+            restart => permanent,
+            type => worker,
+            modules => [seer_server]
+        },
+    supervisor:start_child(seer_sup, Poller).
```

Comment:
This violates the expectations of `application:Module:start/2`; it expects `{ok, Pid}` where `Pid` is the pid of the top-level supervisor. This function appears to return the pid of `seer_server`.

Reply:
This is a very good observation! It's solved as a result of your previous comment about doing that step in the supervisor:init phase.